core-plugin/src/com/google/gct/idea/debugger/CloudBreakpointHandler.java

```diff
@@ -330,13 +349,18 @@ public void registerBreakpoint(@NotNull final XLineBreakpoint<CloudLineBreakpoin
       (CloudLineBreakpointType.CloudLineBreakpoint)cloudIdeBreakpoint;
     if (xIdeBreakpoint.getProperties().isCreatedByServer()) {
       // This newly created IDE breakpoint could be the result of syncing with the server state,
-      // in which case, there is nothing
-      // to do.  Just return.
+      // in which case, there is nothing to do.  Just return.
       // Note that we need to implement this flag as a transient property because this method
       // gets called during construction.
       return;
     }
 
+    if (xIdeBreakpoint.getProperties().isAddedOnServer()) {
```

Comment:
When the breakpoint is added successfully and hit offline, IJ doesn't know it was hit. When you resume the session after the breakpoint is hit, IJ registers every _enabled_ breakpoint with the server. Since IJ didn't know the breakpoint was hit, it didn't mark it as disabled and the breakpoint is therefore registered again.